.github/workflows/scripts/e2e-maven-push.sh

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+# shellcheck source=/dev/null
+source "./.github/workflows/scripts/e2e-utils.sh"
+
+# This script bumps the maven package's version number, commits it, and pushes to
+# the repository.
+
+branch=$(e2e_this_branch)
+
+# NOTE: We can't simply push from $branch because it is occaisonally reset to
+# the main branch. We need to maintain the version number in pom.xml
+# because you cannot overwrite a version in maven. Instead we commit to main,
+# set the tag, reset $branch and push both main and $branch.
+echo "GITHUB_REPOSITORY: ${GITHUB_REPOSITORY}"
+gh repo clone "${GITHUB_REPOSITORY}" -- -b maven-e2e-temp
+repo_name=$(echo "$GITHUB_REPOSITORY" | cut -d '/' -f2)
+cd ./"$repo_name"
+
+git config --global user.name github-actions
+git config --global user.email github-actions@github.com
+
+# Set the remote url to authenticate using the token.
+git remote set-url origin "https://github-actions:${GH_TOKEN}@github.com/${GITHUB_REPOSITORY}.git"
+
+package_dir="${PACKAGE_DIR}" # specified in the e2e test yaml
+
+cd "${package_dir}"
+
+# Get the new version
+artifact_tag=$(mvn help:evaluate -Dexpression=project.version -q -DforceStdout)
+
+# version_major prints the major version number.
+# Expects a string like '1.19.7'
+# version_major returns "1" if the input is '1.19.7'
+version_major() {
+    VER=$(echo $1 | cut -d '.' -f1)
+    echo "$VER"
+}
+
+# version_minor prints the minor version number.
+# Expects a string like '1.19.7'.
+# version_minor returns "19" if the input is '1.19.7'
+version_minor() {
+    VER=$(echo $1 | cut -d '.' -f2)
+    echo "$VER"
+}
+
+# version_patch prints the patch version number.
+# Expects a string like '1.19.7-SNAPSHOT.jar'
+# version_patch returns "7" if the input is '1.19.7'
+version_patch() {
+    VER=$(echo $1 | cut -d '.' -f3)
+    echo "$VER"
+}
+
+# Bumps the version
+new_version() {
+    current_tag=$1
+    release_major=$(version_major "$current_tag")
+    release_minor=$(version_minor "$current_tag")
+    release_patch=$(version_patch "$current_tag")
+
+    # These if-statements are sorted by likelihood
+    if [[ $release_patch != "99" ]]; then
```

Comment:
Is this a requirement for publishing maven packages? What wrong with a version like 1.1.100?

Reply:
It is uncommon practice to go to three digits, but I don't believe the Maven version scheme disallows it.